cartography/driftdetect/cli.py

```diff
@@ -0,0 +1,217 @@
+        )
+        parser.add_argument(
+            '--neo4j-password-prompt',
+            action='store_true',
+            help=(
+                'Present an interactive prompt for a password with which to authenticate to Neo4j. This parameter '
+                'supersedes other methods of supplying a Neo4j password.'
+            ),
+        )
+        parser.add_argument(
+            '-u',
+            '--update',
+            action='store_true',
+            help='Save and updates the results of running the detector query.',
+        )
+        parser.add_argument(
+            '--drift-detector-directory',
+            type=str,
+            default=None,
+            help=(
+                'A path to a directory containing drift-detectors to build. Drift-detection will discover all JSON'
+                'files in the given directory (and its subdirectories) and construct detectors from'
+                'them. Drift-detection does not guarantee the order in which the detector jobs are executed.'
+            ),
+        )
+        return parser
+
+    def configure(self, argv):
+        """
+        Entrypoint for the command line interface.
+
+        :type argv: string
+        :param argv: The parameters supplied to the command line program.
+        """
+        # TODO support parameter lookup in environment variables if not present on command line
+        config = self.parser.parse_args(argv)
+        if config.verbose:
+            logging.getLogger('driftdetect').setLevel(logging.DEBUG)
+        elif config.quiet:
+            logging.getLogger('driftdetect').setLevel(logging.WARNING)
+        else:
+            logging.getLogger('driftdetect').setLevel(logging.INFO)
+        logger.debug("Launching driftdetect with CLI configuration: %r", vars(config))
+        if config.neo4j_user:
+            config.neo4j_password = None
+            if config.neo4j_password_prompt:
+                logger.info("Reading password for Neo4j user '%s' interactively.", config.neo4j_user)
+                config.neo4j_password = getpass.getpass()
+            elif config.neo4j_password_env_var:
+                logger.debug(
+                    "Reading password for Neo4j user '%s' from environment variable '%s'.",
+                    config.neo4j_user,
+                    config.neo4j_password_env_var
+                )
+                config.neo4j_password = os.environ.get(config.neo4j_password_env_var)
+            if not config.neo4j_password:
+                logger.warning("Neo4j username was provided but a password could not be found.")
+        else:
+            config.neo4j_password = None
+        return config
+
+    def main(self, argv):
+        config = self.configure(argv)
+        try:
+            drift_info_detector_pairs = run(config)
+            report_drift(drift_info_detector_pairs)
+        except KeyboardInterrupt:
+            return 130
+
+
+def run(config):
```

Comment:
I think it makes more sense for `run` and `valid_directory` to live in the `detect_drift` module. Consider: you can create your own `Config` instance and invoke `run` through your own code without going through the CLI at all.